pdns/pdns_recursor.cc

```diff
@@ -3777,6 +3777,13 @@ static vector<pair<DNSName, uint16_t> >& operator+=(vector<pair<DNSName, uint16_
   return a;
 }
 
+static pair<int, string>& operator+=(std::pair<int, string>& a, const std::pair<int, string>& b)
```

Comment:
I'm a bit worried about defining an operator for a generic type like that, perhaps defining a new struct would be better?